framework_tvb/tvb/core/neocom/_h5loader.py

```diff
@@ -128,19 +132,26 @@ def load(self, gid):
     def store(self, datatype):
         # type: (HasTraits) -> None
         h5file_cls = self.registry.get_h5file_for_datatype(type(datatype))
-        path = self.path_for(h5file_cls, datatype.gid)
+        if self.base_file is None:
+            path = self.path_for(h5file_cls, datatype.gid)
+        else:
+            path = self.base_file
+            self.base_file = None
```

Comment:
This code is again miss-leading. 
We might have at some point set `self.base_file`, to a not None value, and here, we use that value to store the current file, and regardless if we succeed or not, we set `self.base_file` to None. What are you trying to achieve?
Can't we better extract new methods on this Loader class? Or even create a different Loader subclass?